### ISSUE TITLE
Thread safe version of JUST.net

### ISSUE THREAD

**Issue by RollsRoise** [WorkMaze/JUST.net#81]
It doesn't appear as though this library is thread safe based upon the static variable of GlobalContext in the JsonTransformer class.

We had noticed that occasionally when running async calls, that the result of the transformation was actually from a completely different thread.  After looking at the code, I can see how this might happen based on the the static nature of the GlobalContext.

**Comment by Courela** [comment]
True, attaching input to context was not a smart decision... I guess JsonTransformer will go non-static to accommodate this.

**Comment by RollsRoise** [comment]
@Courela I see that you've made some change to provide thread safety and also your comment in #45 about pull requests not being merged.  Would really love to be able to validate the changes and incorporate - any idea when that might be available?

**Comment by Courela** [comment]
If you are willing to pull the code and compile/use it, you can use [this branch](https://github.com/Truphone/JUST.net/tree/develop).

**Comment by RollsRoise** [comment]
@Courela Thanks!  Preliminary testing looks good - must have picked up the performance improvements too!

**Comment by rahulanand816** [comment]
hi, need some help... I have created an API which tansform  json using just transformer.. having issue using jmeter for testing with multiple thread... occasionally the result of the transformation is being impacted with transformation being performed for a different thread/request

**Comment by Courela** [comment]
You can pass an instance of JUSTContext to Transform method, bypassing the global context, which is static.

**Comment by rahulanand816** [comment]
thanks much @Courela .. I have one question.. I don't have to register custom function even if I use local context.. I still can use those inside transformation json as sharedfunction... Am I Correct?

**Comment by Courela** [comment]
Yes, custom functions can be registered on "Global" context even if you pass a "Local" context to Transform, it searches custom functions in both (local first).
